iwyu.cc

```diff
@@ -2426,9 +2427,16 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
         // parse it to '<new>' before using, so any path that does
         // that, and is clearly a c++ path, is fine; its exact
         // contents don't matter that much.
+
         const FileEntry* use_file = CurrentFileEntry();
-        preprocessor_info().FileInfoFor(use_file)->ReportFullSymbolUse(
-            CurrentLoc(), "<new>", "operator new");
+        const clang::DirectoryLookup* TmpDir = nullptr;
+        auto file = compiler()->getPreprocessor().LookupFile(
```

Comment:
Don't use `auto` unless the type is clearly visible on the line already. `Optional<FileEntryRef>` provides relevant information here.

Reply:
replaced `auto` with the explicit type

Reply:
replaced `auto` with the explicit type